packages/xforms-engine/src/lib/codecs/Geopoint/GeopointValueCodec.ts

```diff
@@ -0,0 +1,46 @@
+import { type CodecDecoder, type CodecEncoder, ValueCodec } from '../ValueCodec.ts';
+import { Geopoint, type GeopointRuntimeValue } from './Geopoint.ts';
+
+// TODO: Add support for GeoJSONValue
+export type GeopointInputValue = GeopointRuntimeValue | string;
+
+const decodeStringValue = (value: GeopointInputValue): GeopointRuntimeValue => {
+	if (typeof value !== 'string' || value.trim() === '') {
+		return null;
+	}
+
+	const [latitude, longitude, altitude = null, accuracy = null] = value.split(/\s+/).map(Number);
+
+	if (latitude == null || longitude == null || Number.isNaN(altitude) || Number.isNaN(accuracy)) {
+		return null;
+	}
+
+	return new Geopoint({ latitude, longitude, altitude, accuracy }).getRuntimeValue();
+};
+
+export class GeopointValueCodec extends ValueCodec<
+	'geopoint',
+	GeopointRuntimeValue,
+	GeopointInputValue
+> {
+	constructor() {
+		const encodeValue: CodecEncoder<GeopointInputValue> = (value) => {
+			const geopointValue = typeof value === 'string' ? decodeStringValue(value) : value;
+
+			if (geopointValue == null) {
+				return '';
+			}
+
+			return new Geopoint(geopointValue)
+				.getTuple()
+				.map((item) => item.value ?? 0)
+				.join(' ');
+		};
+
+		const decodeValue: CodecDecoder<GeopointRuntimeValue> = (value: string) => {
+			return decodeStringValue(value);
+		};
+
+		super('geopoint', encodeValue, decodeValue);
```

Comment:
I'm wondering if it might make sense to move more (maybe even all) of the encode/decode logic to `Geopoint`, e.g. adding:

- `static from(value: Geopoint | string | null): Geopoint | null`
- `serializeXMLValue(): string`[^1]

... with `encodeValue` and `decodeValue` just wrapping those here?

I'm expecting that will be valuable when we go to implement the other geo types, as well as integrate the other geo value use cases.

[^1]: I'm open to bikeshedding the name. I originally typed out `toString` but realized that's error prone and could obscure intent.

Reply:
Done